pkg/controller/factory/controller_context.go

```diff
@@ -107,3 +111,17 @@ func namespaceChecker(interestingNamespaces []string) func(obj interface{}) bool
 		return false
 	}
 }
+
+// rateLimitedAddAfterWorkqueue is like workqueue.RateLimitingInterface, but takes the
```

Comment:
nit: i would move this to separate file, contextually this has nothing to do with context, it is new rate limiter implementation :-)

Reply:
done